packages/api/src/kubernetes-contexts-states.ts

```diff
@@ -0,0 +1,58 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+// CheckingState indicates the state of the check for a context
+export interface CheckingState {
+  state: 'waiting' | 'checking' | 'gaveup';
+}
+
+// A selection of resources, to indicate the 'general' status of a context
+type selectedResources = ['pods', 'deployments'];
+
+// resources managed by podman desktop, excepted the primary ones
+// This is where to add new resources when adding new informers
+export const secondaryResources = [
+  'services',
+  'ingresses',
+  'routes',
+  'configmaps',
+  'secrets',
+  'nodes',
+  'persistentvolumeclaims',
+] as const;
+
+export function isSecondaryResourceName(value: string): value is SecondaryResourceName {
```

Comment:
I don't think we want to have functions exposed in the api packages, moreover this function is not used by any package other than the main ?

Reply:
I also hesitate, I placed it here because it is very close to the `secondaryResourceName` type defined below. I'll move it back to the main package